### PR TITLE
Opcja menu zmieny języka MDZ ,MFI

### PR DESCRIPTION
**Pull request by regit66**

### CODE REVIEW

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
+package com.horstmann.violet.framework.language;
+
+/**
+ * Created by Marcin on 10.01.2017.
+ */
+public class Language {
```
Comment (abizukojc) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java]: The code formatting in project should be like this:
public class Language
{
...
}

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
+package com.horstmann.violet.framework.language;
+
+/**
+ * Created by Marcin on 10.01.2017.
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java]: it should be removed

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+ */
+public class Language {
+
+    private String shortcut;
+    private String name;
+
+    /**
+     * Constructor Language
+     *
+     * @param shortcut, name
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java]: Useless javadoc

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    public Language(String shortcut, String name) {
+        this.shortcut = shortcut;
+        this.name = name;
+    }
+
+    /**
+     * Get language name
+     *
+     * @return name
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java]: as above

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     * Set language name
+     *
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java]: @up

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+     * @param name
+     */
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * Get language shortcut
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java]: @up

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+     *
+     * @return shortcut
+     */
+    public String getShortcut() {
+        return shortcut;
+    }
+
+    /**
+     * Set language shortcut
+     *
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/Language.java]: @up

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java [hunk 8]
```diff
@@ -0,0 +1,93 @@
...
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import com.horstmann.violet.framework.injection.bean.ManiocFramework.InjectedBean;
+import com.horstmann.violet.framework.injection.bean.ManiocFramework.BeanInjector;
+
+/**
+ * Created by Marcin on 10.01.2017.
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java]: It should be removed

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+ */
+public class LanguageManager {
+
+    private List<Language> languages = new ArrayList<Language>();
+    private static final String resourceDir = "com/horstmann/violet/framework/language/";
+    /**
+     * Default constructor
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java]: useless javadoc

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    public LanguageManager() {
+
+        BeanInjector.getInjector().inject(this);
+        loadAvailableLanguage();
+    }
+
+    /**
+     * Get languages List
+     *
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java]: useless javadoc

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+     * @return languages
+     */
+    public List<Language> getLanguages() {
+        return languages;
+    }
+
+    /**
+     * Set PreferedLanguage
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java]: as above

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java [hunk 12]
```diff
@@ -0,0 +1,93 @@
...
+     */
+    public void setPreferedLanguage(String language) {
+        this.userPreferencesServices.setPreferedLanguage(language);
+    }
+
+    /**
+     * Get PreferedLanguage
+     *
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java]: as above

File: violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+
+            }
+
+        }
+
+
+    }
+
+    /**
+     * Apply Prefered Language
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/language/LanguageManager.java]: as above

File: violet-framework/src/main/java/com/horstmann/violet/framework/userpreferences/PreferencesConstant.java [hunk 14]
```diff
@@ -102,8 +102,12 @@ public String toString()
     public static final PreferencesConstant NETWORK_GUESTCONFIG_HTTP_SERVERURL = new PreferencesConstant(
             "network.guestconfig.http.serverurl");
 
-    
     /**
+     +     * Key to store prefered language
```
Comment (nifrez) [inline violet-framework/src/main/java/com/horstmann/violet/framework/userpreferences/PreferencesConstant.java]: remove this plus char

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/MenuFactory_fr.properties [hunk 15]
```diff
@@ -139,7 +139,13 @@ help.homepage.url=http://violet.sourceforge.net/index.php?lang=fr
 help.shortcut.text=Raccourcis
 file.export.error=ERROR
 file.export.error.message=Error writing file\:\n{0}
+setting.text=R�glages
```
Comment (nifrez) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/MenuFactory_fr.properties]: Encoding? use \uXXXXX

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/MenuFactory_fr.properties [hunk 16]
```diff
@@ -139,7 +139,13 @@ help.homepage.url=http://violet.sourceforge.net/index.php?lang=fr
...
+setting.mnemonic=R
+setting.language.text=Langue
+setting.language.mnemonic=L
+setting.language.active.icon=/icons/16x16/fr.png
+setting.dialog.change_language=<html>La langue s�lectionn�e sera <br> activ�e au prochain d�marrage</html>
```
Comment (nifrez) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/MenuFactory_fr.properties]: Encoding? use \uXXXXX

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/SettingMenu.java [hunk 17]
```diff
@@ -0,0 +1,136 @@
...
+
+/**
+ * Represents the setting menu on the editor frame
+ */
+@ResourceBundleBean(resourceReference = MenuFactory.class)
+public class SettingMenu extends JMenu {
+
+    /**
+     * Default constructor
+     *
```
Comment (nifrez) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/SettingMenu.java]: Useless javadoc

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/SettingMenu.java [hunk 18]
```diff
@@ -0,0 +1,136 @@
...
+                                             }
+            );
+
+        }
+
+        this.add(settingItemMenuLanguage);
+    }
+
+    /**
+     * Change language
```
Comment (nifrez) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/SettingMenu.java]: Useless javadoc

File: violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/SettingMenu.java [hunk 19]
```diff
@@ -0,0 +1,136 @@
...
+                null, options, options[0]);
+    }
+
+    /**
+     * Application frame
+     */
+    private MainFrame mainFrame;
+
+    /**
+     * Initialize LanguageManager
```
Comment (nifrez) [inline violetproduct-swing/src/main/java/com/horstmann/violet/application/menu/SettingMenu.java]: Useless javadoc

**Review by nifrez — COMMENTED** [review]
Check comments

**Review by abizukojc — COMMENTED** [review]
Code formatting in all classes should be like this:
```
class Foo 
{
      void bar() 
      {
      }
}
```

**Review by abizukojc — APPROVED** [review]

**Review by nifrez — APPROVED** [review]

**Review by orchowski — APPROVED** [review]

**Pull request merged**